### ISSUE TITLE
MathLoadTest_all_special doesn't timeout and collect diagnostics

### ISSUE THREAD

**Issue by pshipton** [AdoptOpenJDK/openjdk-systemtest#389]
https://ci.eclipse.org/openj9/job/Test_openjdk8_j9_special.system_ppc64_aix_Nightly_testList_4/34
MathLoadTest_all_special_17
The test appears to hang, but STF doesn't timeout and collect any diagnostics, instead running for hours.
```
23:44:16  MLT 23:44:16.525 - Completed 9.7%. Number of tests started=507 (+55)
23:44:39  MLT 23:44:36.580 - Completed 10.0%. Number of tests started=523 (+16)
23:44:58  MLT 23:44:56.567 - Completed 12.5%. Number of tests started=658 (+135)
23:45:15  STF 23:45:14.023 - Heartbeat: Process MLT is still running
23:45:17  MLT 23:45:16.587 - Completed 15.0%. Number of tests started=787 (+129)
23:50:16  STF 23:50:14.149 - Heartbeat: Process MLT is still running
23:55:26  STF 23:55:14.208 - Heartbeat: Process MLT is still running
00:00:23  STF 00:00:14.296 - Heartbeat: Process MLT is still running
00:05:18  STF 00:05:14.400 - Heartbeat: Process MLT is still running
00:10:29  STF 00:10:14.453 - Heartbeat: Process MLT is still running
00:15:24  STF 00:15:14.498 - Heartbeat: Process MLT is still running
00:20:20  STF 00:20:14.034 - Heartbeat: Process MLT is still running
00:25:28  STF 00:25:14.089 - Heartbeat: Process MLT is still running
00:30:21  STF 00:30:14.149 - Heartbeat: Process MLT is still running
00:35:15  STF 00:35:14.189 - Heartbeat: Process MLT is still running
00:40:26  STF 00:40:14.267 - Heartbeat: Process MLT is still running
00:45:28  STF 00:45:14.343 - Heartbeat: Process MLT is still running
00:50:29  STF 00:50:14.495 - Heartbeat: Process MLT is still running
00:55:29  STF 00:55:14.097 - Heartbeat: Process MLT is still running
01:00:28  STF 01:00:14.229 - Heartbeat: Process MLT is still running
01:05:18  STF 01:05:14.370 - Heartbeat: Process MLT is still running
01:10:18  STF 01:10:14.473 - Heartbeat: Process MLT is still running
01:15:20  STF 01:15:14.036 - Heartbeat: Process MLT is still running
01:20:26  STF 01:20:14.101 - Heartbeat: Process MLT is still running
01:25:28  STF 01:25:14.175 - Heartbeat: Process MLT is still running
01:30:30  STF 01:30:14.233 - Heartbeat: Process MLT is still running
01:35:29  STF 01:35:14.296 - Heartbeat: Process MLT is still running
01:40:27  STF 01:40:14.360 - Heartbeat: Process MLT is still running
01:45:17  STF 01:45:14.441 - Heartbeat: Process MLT is still running
01:50:18  STF 01:50:14.036 - Heartbeat: Process MLT is still running
01:55:24  STF 01:55:14.132 - Heartbeat: Process MLT is still running
02:00:15  STF 02:00:14.210 - Heartbeat: Process MLT is still running
02:05:26  STF 02:05:14.294 - Heartbeat: Process MLT is still running
02:10:20  STF 02:10:14.377 - Heartbeat: Process MLT is still running
02:15:16  STF 02:15:14.466 - Heartbeat: Process MLT is still running
02:20:22  STF 02:20:14.047 - Heartbeat: Process MLT is still running
02:25:16  STF 02:25:14.135 - Heartbeat: Process MLT is still running
02:30:24  STF 02:30:14.236 - Heartbeat: Process MLT is still running
02:35:17  STF 02:35:14.319 - Heartbeat: Process MLT is still running
02:40:20  STF 02:40:14.412 - Heartbeat: Process MLT is still running
02:45:17  STF 02:45:14.497 - Heartbeat: Process MLT is still running
02:50:28  STF 02:50:14.106 - Heartbeat: Process MLT is still running
02:55:27  STF 02:55:14.185 - Heartbeat: Process MLT is still running
03:00:24  STF 03:00:14.247 - Heartbeat: Process MLT is still running
03:05:22  STF 03:05:14.274 - Heartbeat: Process MLT is still running
03:10:24  STF 03:10:14.347 - Heartbeat: Process MLT is still running
03:15:15  STF 03:15:14.419 - Heartbeat: Process MLT is still running
03:20:24  STF 03:20:14.037 - Heartbeat: Process MLT is still running
03:25:27  STF 03:25:14.165 - Heartbeat: Process MLT is still running
03:30:16  STF 03:30:14.282 - Heartbeat: Process MLT is still running
03:35:41  STF 03:35:14.372 - Heartbeat: Process MLT is still running
03:40:23  STF 03:40:14.480 - Heartbeat: Process MLT is still running
03:45:16  STF 03:45:14.145 - Heartbeat: Process MLT is still running
03:46:40  Cannot contact aix71-p8-6: java.lang.InterruptedException
03:50:21  STF 03:50:14.236 - Heartbeat: Process MLT is still running
03:55:17  STF 03:55:14.328 - Heartbeat: Process MLT is still running
04:00:22  STF 04:00:14.435 - Heartbeat: Process MLT is still running
04:05:19  STF 04:05:14.084 - Heartbeat: Process MLT is still running
04:10:27  STF 04:10:14.177 - Heartbeat: Process MLT is still running
04:15:24  STF 04:15:14.295 - Heartbeat: Process MLT is still running
04:20:18  STF 04:20:14.357 - Heartbeat: Process MLT is still running
04:25:22  STF 04:25:14.412 - Heartbeat: Process MLT is still running
04:30:16  STF 04:30:14.487 - Heartbeat: Process MLT is still running
04:35:20  STF 04:35:14.032 - Heartbeat: Process MLT is still running
04:40:23  STF 04:40:14.099 - Heartbeat: Process MLT is still running
04:45:17  STF 04:45:14.160 - Heartbeat: Process MLT is still running
04:50:28  STF 04:50:14.254 - Heartbeat: Process MLT is still running
04:55:21  STF 04:55:14.348 - Heartbeat: Process MLT is still running
05:00:29  STF 05:00:14.446 - Heartbeat: Process MLT is still running
05:05:22  STF 05:05:14.026 - Heartbeat: Process MLT is still running
05:10:14  STF 05:10:14.105 - Heartbeat: Process MLT is still running
05:15:22  STF 05:15:14.174 - Heartbeat: Process MLT is still running
05:20:15  STF 05:20:14.253 - Heartbeat: Process MLT is still running
05:25:23  STF 05:25:14.317 - Heartbeat: Process MLT is still running
05:30:16  STF 05:30:14.333 - Heartbeat: Process MLT is still running
05:35:24  STF 05:35:14.428 - Heartbeat: Process MLT is still running
05:40:16  STF 05:40:14.009 - Heartbeat: Process MLT is still running
05:45:24  STF 05:45:14.057 - Heartbeat: Process MLT is still running
05:50:17  STF 05:50:14.095 - Heartbeat: Process MLT is still running
05:55:25  STF 05:55:14.128 - Heartbeat: Process MLT is still running
06:00:18  STF 06:00:14.178 - Heartbeat: Process MLT is still running
06:05:28  STF 06:05:14.260 - Heartbeat: Process MLT is still running
06:10:21  STF 06:10:14.328 - Heartbeat: Process MLT is still running
06:15:29  STF 06:15:14.409 - Heartbeat: Process MLT is still running
06:20:22  STF 06:20:14.486 - Heartbeat: Process MLT is still running
06:25:16  STF 06:25:14.061 - Heartbeat: Process MLT is still running
06:30:27  STF 06:30:14.104 - Heartbeat: Process MLT is still running
06:35:23  STF 06:35:14.151 - Heartbeat: Process MLT is still running
06:40:19  STF 06:40:14.230 - Heartbeat: Process MLT is still running
06:45:15  STF 06:45:14.312 - Heartbeat: Process MLT is still running
06:50:26  STF 06:50:14.382 - Heartbeat: Process MLT is still running
06:55:21  STF 06:55:14.478 - Heartbeat: Process MLT is still running
07:00:17  STF 07:00:14.049 - Heartbeat: Process MLT is still running
07:05:28  STF 07:05:14.093 - Heartbeat: Process MLT is still running
07:10:23  STF 07:10:14.066 - Heartbeat: Process MLT is still running
07:15:18  STF 07:15:14.053 - Heartbeat: Process MLT is still running
07:20:29  STF 07:20:14.044 - Heartbeat: Process MLT is still running
07:25:24  STF 07:25:14.483 - Heartbeat: Process MLT is still running
07:30:19  STF 07:30:14.478 - Heartbeat: Process MLT is still running
07:35:15  STF 07:35:14.470 - Heartbeat: Process MLT is still running
07:40:26  STF 07:40:14.487 - Heartbeat: Process MLT is still running
07:45:20  STF 07:45:14.032 - Heartbeat: Process MLT is still running
07:50:14  STF 07:50:14.054 - Heartbeat: Process MLT is still running
07:55:22  STF 07:55:14.060 - Heartbeat: Process MLT is still running
08:00:15  STF 08:00:14.039 - Heartbeat: Process MLT is still running
08:05:25  STF 08:05:14.012 - Heartbeat: Process MLT is still running
08:10:18  STF 08:10:14.022 - Heartbeat: Process MLT is still running
08:15:26  STF 08:15:14.078 - Heartbeat: Process MLT is still running
08:20:18  STF 08:20:14.099 - Heartbeat: Process MLT is still running
08:25:26  STF 08:25:14.064 - Heartbeat: Process MLT is still running
08:30:19  STF 08:30:14.000 - Heartbeat: Process MLT is still running
08:35:26  STF 08:35:14.471 - Heartbeat: Process MLT is still running
08:40:19  STF 08:40:14.435 - Heartbeat: Process MLT is still running
08:45:27  STF 08:45:14.402 - Heartbeat: Process MLT is still running
08:50:19  STF 08:50:14.361 - Heartbeat: Process MLT is still running
08:55:28  STF 08:55:14.334 - Heartbeat: Process MLT is still running
09:00:21  STF 09:00:14.294 - Heartbeat: Process MLT is still running
Cancelling nested steps due to timeout
```

**Comment by pshipton** [comment]
Another one https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_special.system_ppc64le_linux_Nightly_testList_4/32 (MathLoadTest_all_special_22)
It hasn't hung, it's still running and may complete,, but started at 04:17:31 and still going at 09:27:00. The test should timeout rather than running for this long.

**Comment by pshipton** [comment]
See also https://github.com/eclipse/openj9/issues/11190. I suspect these tests aren't well configured for these modes (17 and 22). Probably https://github.com/AdoptOpenJDK/openjdk-tests/issues/2104 is the answer.

**Comment by Mesbah-Alam** [comment]
STF is working as designed here. The problem is we have set a timeout of `20h` for this test: 

```
04:17:35  STF 05:12:23.601 - |   Expectation: CLEAN_RUN within 20h
```

The test starts at `04:17:31 ` and ends successfully at `09:43:38`. So, it takes almost 5.5 hrs to complete in [this](https://ci.eclipse.org/openj9/job/Test_openjdk11_j9_special.system_ppc64le_linux_Nightly_testList_4/32/consoleFull) particular build. 

How long is test supposed to take to complete using the special modes that it is using? ("-Xcompressedrefs -Xgcpolicy:gencon -Xjit:counts="- - - - - - 1 1 1 1000 250 250 - - - 10000 100000 10000",gcOnResolve,rtResolve,sampleInterval=2,scorchingSampleThreshold=10000,quickProfile").   

We can reduce the timeout, but then this test will obviously fail due to timeout.

**Comment by Mesbah-Alam** [comment]
There are a few issues in MathLoadTest: 

1) We should have a more realistic timeout value for the case where the test specifies `20h` for timeout:  https://github.com/AdoptOpenJDK/openjdk-systemtest/blob/master/openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MathLoadTest.java#L37 

2) We had [hardcoded](https://github.com/AdoptOpenJDK/openjdk-systemtest/blob/master/openjdk.test.load/src/test.load/net/adoptopenjdk/stf/MathLoadTest.java#L90) some JIT options in the MathLoadTest so that, if any of those JIT options is seen in the command line, then that particular instance of the MathLoadTest will be treated as a potentially "slow running" one and a longer timeout values will be set for it.  

Problem with this approach is when we add new modes in the playlist, we end up with new JIT options which are potentially slow running too, but are not present in the hard coded list of JIT options in the test. So the test does not treat them as "slow running" cases. We need to find a better solution for this.

**Comment by pshipton** [comment]
My preferred solution is to roll out https://github.com/AdoptOpenJDK/openjdk-tests/issues/2104 for all system testing, although not necessarily all at the same time. This will stop extra long running tests from consuming too much time, and start running short running tests longer, increasing the changes of finding bugs.

**Comment by pshipton** [comment]
> How long is test supposed to take to complete using the special modes that it is using? 

It's not possible to answer this. It depends on the machine speed, machine virtualization (how processors are shared and the load across all the instances), the speed may vary from run to run depending on timing and random factors, and in theory could get faster over time as performance improvements are delivered. I assume the test is running some number of iterations, so the answer also depends on how many iterations are being run. 

Bottom line is we don't want to run these tests for 5 hours, or even 30min in a nightly build. Historically we've chosen 5 min, so I expect we should start with that target. Once all testing is adjusted to do this, if there is enough machine capacity we could consider a longer time like 10min. Perhaps in the future we'll also run longer tests (1 hour?) more infrequently.

**Comment by Mesbah-Alam** [comment]
New workload and timeout settings delivered for MathLoadTest via https://github.com/eclipse/openj9/issues/11666 should resolve this issue as well.
